test/infra/runner.sh

```
@@ -87,24 +87,75 @@ setup_job_watch "${JOB_NAME_PM}" "PM_TEST"
 SM_JOB_STATUS=-1 # -1: running, 0: success, 1: failed
 PM_JOB_STATUS=-1 # -1: running, 0: success, 1: failed
 
+# Helper function to get detailed info about a job and its pods on failure or timeout
+get_job_info_on_failure() {
+    local job_name=$1
+    local namespace="e2e-test" # Assuming fixed namespace from context
+    echo "---"
+    echo "Gathering diagnostic information for job '$job_name' in namespace '$namespace'..."
+    echo "---"
+
+    echo "Describing job '$job_name':"
+    kubectl describe job "$job_name" -n "$namespace"
+    echo "---"
+
+    echo "Getting pods for job '$job_name':"
+    # Using jsonpath to get .metadata.name directly, handles multiple pods
+    pod_names=$(kubectl get pods -n "$namespace" -l "job-name=$job_name" -o jsonpath='{.items[*].metadata.name}')
+
+    if [ -z "$pod_names" ]; then
+        echo "No pods found for job '$job_name'."
+    else
+        for pod_name in $pod_names; do
+            echo "" # Add a newline for better separation before each pod's info
+            echo "Describing pod '$pod_name':"
+            kubectl describe pod "$pod_name" -n "$namespace"
+            echo "---"
+            echo "Logs for pod '$pod_name' (last 200 lines, with prefix):"
+            # --prefix is useful if the pod has multiple containers or if you're aggregating logs
+            kubectl logs "$pod_name" -n "$namespace" --tail=200 --prefix
+            echo "---"
+        done
+    fi
+    echo "Finished gathering diagnostic information for job '$job_name'."
+    echo "---"
+}
+
 # Function to check job status
 # Arguments: $1=job_name
+# Returns 0 for success, 1 for failure.
```

The condition below also needs update, as we never send 255

        if [ $ret -ne 255 ]; then

---

If the job completes successfully within 45 minutes, kubectl wait exits successfully, and check_job_status returns 0.
If the job does not complete successfully within 45 minutes (either it fails explicitly, or the kubectl wait command times out), check_job_status considers this a failure for that job's lifecycle and returns 1.



Each call to check_job_status will now give a definitive outcome instead of blocking wait using the while loop, after waiting up to the completion_timeout.


Since check_job_status never returns 255 now, as you pointed out it will either return 0 or 1

-----------
Exit code
----------
0 -> success
1 -> fail

Have simplified the if block slightly(avoided it altogether 😅)